annotation/tests/test_annotations.py

```diff
@@ -482,3 +590,571 @@ def test_construct_annotated_doc_db_error(
+        mock_load_page.assert_has_calls(
+            [
+                call(
+                    None,
+                    [],
+                    TEST_TENANT,
+                    1,
+                    expected_user_1,
+                    expected_page_rev_1,
+                    True,
+                ),
+                call(
+                    None,
+                    [],
+                    TEST_TENANT,
+                    2,
+                    expected_user_2,
+                    expected_page_rev_2,
+                    True,
+                ),
+            ]
+        )
+        assert mock_load_page.call_count == 2
+    assert page_revision_list_latest == expected_pages
+
+
+def test_load_annotated_pages_for_particular_rev(
+    annotated_doc: AnnotatedDoc,
+    page_revision_list_all: Dict[int, List[PageRevision]],
+):
+    annotated_doc.pages[1] = "sha1"
+    with patch(
+        "annotation.annotations.main.load_page"
+    ) as mock_load_page, patch(
+        "annotation.annotations.main.logger_.debug"
+    ) as mock_debug:
+        load_annotated_pages_for_particular_rev(
+            annotated_doc, page_revision_list_all[1][0], TEST_TENANT, []
+        )
+        mock_debug.assert_called_once_with(
+            "load_annotated_pages_for_particular_rev"
+        )
+        mock_load_page.assert_called_once_with(
+            None,
+            [],
+            TEST_TENANT,
+            1,
+            annotated_doc.user,
+            page_revision_list_all[1][0],
+            False,
+        )
+    assert page_revision_list_all[1][0]["page_id"] == "sha1"
+
+
+def test_load_validated_pages_for_particular_rev(
+    annotated_doc: AnnotatedDoc,
+    page_revision_list_all: Dict[int, List[PageRevision]],
+):
+    with patch(
+        "annotation.annotations.main.convert_bucket_name_if_s3prefix",
+        return_value=TEST_TENANT,
+    ) as mock_convert_prefix, patch(
+        "annotation.annotations.main.load_page"
+    ) as mock_load_page, patch(
+        "annotation.annotations.main.logger_.debug"
+    ) as mock_debug:
+        load_validated_pages_for_particular_rev(
+            annotated_doc, page_revision_list_all[1][0], TEST_TENANT, []
+        )
+        mock_debug.assert_called_once_with(
+            "load_validated_pages_for_particular_rev"
+        )
+        mock_convert_prefix.assert_called_once_with(TEST_TENANT)
+        mock_load_page.assert_called_once_with(
+            None,
+            [],
+            TEST_TENANT,
+            1,
+            annotated_doc.user,
+            page_revision_list_all[1][0],
+            False,
+        )
+    assert page_revision_list_all[1][0]["page_id"] is None
+
+
+def test_construct_particular_rev_response(annotated_doc: AnnotatedDoc):
+    page = {
+        "page_num": 1,
+        "size": {"width": 6.0, "height": 9.0},
+        "objs": [],
+    }
+
+    def mock_loaded_pages_append(
+        revision, page_revision, tenant, loaded_pages
```

Comment:
Add types

Reply:
Perhaps we should consider removing revision, page_revision, and tenant since they are not used in this nested function.

Reply:
Added typing. It needs to have all four arguments, since the call in the function has 4 parameters.